### DISCUSSION TITLE
Any plans on supporting SparseML?

### DISCUSSION THREAD

**Discussion by tranlm** [microsoft/onnxruntime-genai#902] [Ideas]
I was looking for libraries that can prune in addition to quantizing and it appears that SparseML can do that. However, I tried taking the onnx weights exported by the library and loading it to genai. I keep getting errors like the one below. I'm guessing that maybe pruning isn't supported by this library yet?

```
Traceback (most recent call last):
  File "<stdin>", line 3, in <module>
onnxruntime_genai.onnxruntime_genai.OrtException: Got invalid dimensions for input: input_ids for the following indices
 index: 1 Got: 6 Expected: 4096
 Please fix either the inputs/outputs or the model.
 ```

**Comment by RyanUnderhill** [comment]
Can you give more information about what you tried? What model were you using? This error is coming from onnxruntime, vs genai itself. It appears the model expects input_ids to be 4096 in size and 6 is passed in. Is this expecting input_ids to be padded out to 4096?
